### DISCUSSION TITLE
Timeout when running hover_swarm.launch

### DISCUSSION THREAD

**Discussion by armrog** [USC-ACTLab/crazyswarm#521] [Q&A]
Hi, 
We are having problems when trying to launch hover_swarm.launch. We are using qualisys with libobjecttracker and we can read our position when launching mocap_helper. When using chooser.py we can see the battery level and reboot the drone successfully. Therefore we assume that we have a connection to the drone as well as the motion capture system. Both the crazy radio and the drone is running the latest software. Below is the output from the terminal when running hover_swarm.launch.

We saw that someone had the same issue but it was not resolved, do you know what is causing the error or do you have any suggestions on how to trouble shoot the problem? 
It would be greatly appreciated!

`fs-as2@fsas2-HP-ProBook-640-G1:~/tsfs12/crazyswarm$ roslaunch crazyswarm hover_swarm.launch 
... logging to /home/fs-as2/.ros/log/7ba8b3e2-3c84-11ec-8e10-c4d9879035e4/roslaunch-fsas2-HP-ProBook-640-G1-6543.log
Checking log directory for disk usage. This may take a while.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

started roslaunch server http://fsas2-HP-ProBook-640-G1:45421/

SUMMARY
========

PARAMETERS
 * /crazyflieTypes/box/batteryVoltageWarning: 3.8
 * /crazyflieTypes/box/batteryVoltateCritical: 3.7
 * /crazyflieTypes/box/bigQuad: True
 * /crazyflieTypes/box/dynamicsConfiguration: 0
 * /crazyflieTypes/box/firmwareParams/ctrlMel/i_range_m_z: 1500
 * /crazyflieTypes/box/firmwareParams/ctrlMel/i_range_xy: 2.0
 * /crazyflieTypes/box/firmwareParams/ctrlMel/i_range_z: 0.4
 * /crazyflieTypes/box/firmwareParams/ctrlMel/kR_xy: 70000
 * /crazyflieTypes/box/firmwareParams/ctrlMel/kR_z: 60000
 * /crazyflieTypes/box/firmwareParams/ctrlMel/kd_omega_rp: 200
 * /crazyflieTypes/box/firmwareParams/ctrlMel/kd_xy: 0.2
 * /crazyflieTypes/box/firmwareParams/ctrlMel/kd_z: 0.4
 * /crazyflieTypes/box/firmwareParams/ctrlMel/ki_m_z: 500
 * /crazyflieTypes/box/firmwareParams/ctrlMel/ki_xy: 0.05
 * /crazyflieTypes/box/firmwareParams/ctrlMel/ki_z: 0.05
 * /crazyflieTypes/box/firmwareParams/ctrlMel/kp_xy: 0.4
 * /crazyflieTypes/box/firmwareParams/ctrlMel/kp_z: 1.25
 * /crazyflieTypes/box/firmwareParams/ctrlMel/kw_xy: 20000
 * /crazyflieTypes/box/firmwareParams/ctrlMel/kw_z: 12000
 * /crazyflieTypes/box/firmwareParams/ctrlMel/mass: 0.032
 * /crazyflieTypes/box/firmwareParams/ctrlMel/massThrust: 132000
 * /crazyflieTypes/box/markerConfiguration: 4
 * /crazyflieTypes/default/batteryVoltageWarning: 3.8
 * /crazyflieTypes/default/batteryVoltateCritical: 3.7
 * /crazyflieTypes/default/bigQuad: False
 * /crazyflieTypes/default/dynamicsConfiguration: 0
 * /crazyflieTypes/default/firmwareParams/ctrlMel/i_range_m_z: 1500
 * /crazyflieTypes/default/firmwareParams/ctrlMel/i_range_xy: 2.0
 * /crazyflieTypes/default/firmwareParams/ctrlMel/i_range_z: 0.4
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kR_xy: 70000
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kR_z: 60000
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kd_omega_rp: 200
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kd_xy: 0.2
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kd_z: 0.4
 * /crazyflieTypes/default/firmwareParams/ctrlMel/ki_m_z: 500
 * /crazyflieTypes/default/firmwareParams/ctrlMel/ki_xy: 0.05
 * /crazyflieTypes/default/firmwareParams/ctrlMel/ki_z: 0.05
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kp_xy: 0.4
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kp_z: 1.25
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kw_xy: 20000
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kw_z: 12000
 * /crazyflieTypes/default/firmwareParams/ctrlMel/mass: 0.032
 * /crazyflieTypes/default/firmwareParams/ctrlMel/massThrust: 132000
 * /crazyflieTypes/default/markerConfiguration: 0
 * /crazyflieTypes/large/batteryVoltageWarning: 11.4
 * /crazyflieTypes/large/batteryVoltateCritical: 11.1
 * /crazyflieTypes/large/bigQuad: True
 * /crazyflieTypes/large/dynamicsConfiguration: 0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/i_range_m_z: 1500
 * /crazyflieTypes/large/firmwareParams/ctrlMel/i_range_xy: 2.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/i_range_z: 4.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kR_xy: 30000
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kR_z: 40000
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kd_omega_rp: 100
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kd_xy: 3.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kd_z: 3.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/ki_m_z: 500
 * /crazyflieTypes/large/firmwareParams/ctrlMel/ki_xy: 1.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/ki_z: 2.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kp_xy: 8.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kp_z: 8.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kw_xy: 9000
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kw_z: 10000
 * /crazyflieTypes/large/firmwareParams/ctrlMel/mass: 0.38
 * /crazyflieTypes/large/firmwareParams/ctrlMel/massThrust: 5400
 * /crazyflieTypes/large/markerConfiguration: 2
 * /crazyflieTypes/medium/batteryVoltageWarning: 7.6
 * /crazyflieTypes/medium/batteryVoltateCritical: 7.4
 * /crazyflieTypes/medium/bigQuad: True
 * /crazyflieTypes/medium/dynamicsConfiguration: 0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/i_range_m_z: 1500
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/i_range_xy: 2.0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/i_range_z: 2.0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kR_xy: 50000
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kR_z: 60000
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kd_omega_rp: 200
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kd_xy: 1.0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kd_z: 1.0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/ki_m_z: 500
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/ki_xy: 0.1
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/ki_z: 0.5
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kp_xy: 2.0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kp_z: 3
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kw_xy: 16000
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kw_z: 12000
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/mass: 0.11
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/massThrust: 23000
 * /crazyflieTypes/medium/markerConfiguration: 1
 * /crazyflies: [{'initialPositio...
 * /crazyswarm_server/broadcast_address: FFE7E7E7E7
 * /crazyswarm_server/enable_logging: False
 * /crazyswarm_server/enable_parameters: True
 * /crazyswarm_server/firmware: crazyswarm
 * /crazyswarm_server/firmwareParams/commander/enHighLevel: 1
 * /crazyswarm_server/firmwareParams/ring/effect: 16
 * /crazyswarm_server/firmwareParams/ring/headlightEnable: 0
 * /crazyswarm_server/firmwareParams/ring/solidBlue: 255
 * /crazyswarm_server/firmwareParams/ring/solidGreen: 0
 * /crazyswarm_server/firmwareParams/ring/solidRed: 0
 * /crazyswarm_server/firmwareParams/stabilizer/controller: 2
 * /crazyswarm_server/firmwareParams/stabilizer/estimator: 3
 * /crazyswarm_server/force_no_cache: False
 * /crazyswarm_server/genericLogTopicFrequencies: [10]
 * /crazyswarm_server/genericLogTopic_log1_Variables: ['ctrlStat.edist']
 * /crazyswarm_server/genericLogTopics: ['log1']
 * /crazyswarm_server/motion_capture_type: qualisys
 * /crazyswarm_server/object_tracking_type: libobjecttracker
 * /crazyswarm_server/print_latency: False
 * /crazyswarm_server/qualisys_base_port: 22222
 * /crazyswarm_server/qualisys_host_name: 192.168.0.50
 * /crazyswarm_server/save_point_clouds: ~/pointCloud.ot
 * /crazyswarm_server/world_frame: /world
 * /crazyswarm_server/write_csvs: False
 * /crazyswarm_teleop/csv_file: /home/fs-as2/craz...
 * /crazyswarm_teleop/timescale: 0.8
 * /dynamicsConfigurations/0/maxFitnessScore: 0.001
 * /dynamicsConfigurations/0/maxPitch: 1.4
 * /dynamicsConfigurations/0/maxPitchRate: 20.0
 * /dynamicsConfigurations/0/maxRoll: 1.4
 * /dynamicsConfigurations/0/maxRollRate: 20.0
 * /dynamicsConfigurations/0/maxXVelocity: 2.0
 * /dynamicsConfigurations/0/maxYVelocity: 2.0
 * /dynamicsConfigurations/0/maxYawRate: 10.0
 * /dynamicsConfigurations/0/maxZVelocity: 3.0
 * /markerConfigurations/0/numPoints: 4
 * /markerConfigurations/0/offset: [0.0, 0.0, -0.03]
 * /markerConfigurations/0/points/0: [-0.0030885, -0.0...
 * /markerConfigurations/0/points/1: [0.0369053, -0.00...
 * /markerConfigurations/0/points/2: [0.00691818, -0.0...
 * /markerConfigurations/0/points/3: [0.00681874, 0.02...
 * /markerConfigurations/1/numPoints: 4
 * /markerConfigurations/1/offset: [0.0, 0.0, -0.03]
 * /markerConfigurations/1/points/0: [-0.00896228, -0....
 * /markerConfigurations/1/points/1: [-0.0156318, 0.09...
 * /markerConfigurations/1/points/2: [0.0461693, -0.08...
 * /markerConfigurations/1/points/3: [-0.0789959, -0.0...
 * /markerConfigurations/2/numPoints: 4
 * /markerConfigurations/2/offset: [0.0, 0.0, -0.06]
 * /markerConfigurations/2/points/0: [0.0558163, -0.00...
 * /markerConfigurations/2/points/1: [-0.0113941, 0.00...
 * /markerConfigurations/2/points/2: [-0.0306277, 0.05...
 * /markerConfigurations/2/points/3: [0.0535816, -0.04...
 * /markerConfigurations/3/numPoints: 1
 * /markerConfigurations/3/offset: [0.0, -0.01, -0.04]
 * /markerConfigurations/3/points/0: [0.0177184, 0.013...
 * /markerConfigurations/4/numPoints: 4
 * /markerConfigurations/4/offset: [0.0, 0.0, 0.0]
 * /markerConfigurations/4/points/0: [0.114936, -0.016...
 * /markerConfigurations/4/points/1: [-0.0462025, 0.01...
 * /markerConfigurations/4/points/2: [0.0383504, -0.01...
 * /markerConfigurations/4/points/3: [-0.0472455, -0.0...
 * /numDynamicsConfigurations: 1
 * /numMarkerConfigurations: 5
 * /rosdistro: melodic
 * /rosversion: 1.14.9

NODES
  /
    crazyswarm_server (crazyswarm/crazyswarm_server)
    crazyswarm_teleop (crazyswarm/crazyswarm_teleop)
    rviz (rviz/rviz)

auto-starting new master
process[master]: started with pid [6553]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to 7ba8b3e2-3c84-11ec-8e10-c4d9879035e4
process[rosout-1]: started with pid [6564]
started core service [/rosout]
process[crazyswarm_server-2]: started with pid [6571]
process[crazyswarm_teleop-3]: started with pid [6572]
process[rviz-4]: started with pid [6573]
[ INFO] [1635930025.914287983]: Wait for services...
[ INFO] [1635930025.915857611]: waitForService: Service [/emergency] has not been advertised, waiting...
[ INFO] [1635930025.977977100]: waitForService: Service [/emergency] is now available.
[ INFO] [1635930025.979879489]: Manager ready.
ch: 1
terminate called after throwing an instance of 'std::runtime_error'
  what():  timeout
[crazyswarm_server-2] process has died [pid 6571, exit code -6, cmd /home/fs-as2/tsfs12/crazyswarm/ros_ws/devel/lib/crazyswarm/crazyswarm_server __name:=crazyswarm_server __log:=/home/fs-as2/.ros/log/7ba8b3e2-3c84-11ec-8e10-c4d9879035e4/crazyswarm_server-2.log].
log file: /home/fs-as2/.ros/log/7ba8b3e2-3c84-11ec-8e10-c4d9879035e4/crazyswarm_server-2*.log
[rviz-4] process has finished cleanly
log file: /home/fs-as2/.ros/log/7ba8b3e2-3c84-11ec-8e10-c4d9879035e4/rviz-4*.log
^C[crazyswarm_teleop-3] killing on exit
[rosout-1] killing on exit
[master] killing on exit
shutting down processing monitor...
... shutting down processing monitor complete
done
`

**Comment by whoenig** [comment]
This usually indicates that the communication to the CF(s) doesn't work. Could you provide the content of `crazyflies.yaml` as well as the URI(s) of your Crazyflie(s)?

**Reply by armrog** [reply]
Thank you so much for the quick response.

We also suspected that it was the connection with the CF that was the problem. Therefore changed to the 2M bandwidth instead of the 250K aswell as updated the CF. We are currently only running one CF with the URI E7E7E7E701. Only drone 1 is selected in chooser.py and as we said before we managed to get battery level from chooser.py. But we are still having the timeout problem. Below is the crazyflies.yaml. 



 here is the crazyflies.yaml

`crazyflies:
- channel: 100
  id: 1
  initialPosition: [1.0, 0.0, 0.0]
  type: default
- channel: 100
  id: 2
  initialPosition: [0.0, 0.0, 0.0]
  type: default`

**Comment by armrog** [comment]
Hello again! small update, we tried with a brand new flashed crazyflie 2.1 instead of the 2.0 we used before. However we still have the same problems as before.

**Reply by whoenig** [reply]
What is your full URI? The Crazyswarm only supports the 2M bandwidth. Once you have the URI, you can also try the crazyflie_tools, e.g., `rosrun crazyflie_tools scan --address E7E7E7E701` should give you your URI, other tools like `listLogVariables` should be able to tell you if a connection can be established.

**Reply by armrog** [reply]
It works! 
It was as you said, a problem with the communication with the CF. 
We are new to ros in general and we did not realize how chooser.py, allCrazyflies.yaml and crazyflie were connected at fist. After your answer we knew where to look for the problem and managed to fix it. 
Thanks
